### PR TITLE
Defunct expand_model_out_val_grid and create_model_out_submit_tmpl and move to hubValidations

### PR DESCRIPTION
**Pull request by annakrystalli**
This PR introduces the following  breaking change: `expand_model_out_val_grid()` and `create_model_out_submit_tmpl()` are now defunt. These functions have been moved to `hubValidations` and replaced by `hubValidations::expand_model_out_grid()` and `hubValidations::submission_tmpl()`, respectively. The old functions will now error if called with a message that they have moved to `hubValidations` and will be removed in a future release.

`hubValidations` is a much more natural place for these functions as `expand_model_out_val_grid()` is exclusively used internally by `hubValidation` check functions while `create_model_out_submit_tmpl` seems to also fit better as a helper for creating template files that can then becompleted by teams ams validated locally before submission. Hence `hubValidations` feels like the appropriate home for this function too.

This PR and is related to the associated PR in `hubValidations`  https://github.com/hubverse-org/hubValidations/pull/95

### CODE REVIEW

File: NEWS.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
+# hubData 1.0.0
+
+* Breaking change: `expand_model_out_val_grid()` and `create_model_out_submit_tmpl()` are now defunt. These functions have been moved to `hubValidations` and replaced by `hubValidations::expand_model_out_grid()` and `hubValidations::submission_tmpl()`, respectively. The old functions will now fail if called and will be removed in a future release.
```
Comment (nickreich) [inline NEWS.md:3]: "defunt" should be "defunct"

File: R/hubData-package.R
Comment (nickreich) [inline R/hubData-package.R:1]: not clear to me what these new lines add, but they seem auto-generated and tied to the use of `lifecycle` so I assume it's ok

**Review by nickreich — APPROVED** [review]
One minor typo change requested, otherwise LGTM.

File: NEWS.md (hunk 1 above)
Comment (annakrystalli) [reply NEWS.md]: Doh! Thanks! Fixed now.

File: R/hubData-package.R
Comment (annakrystalli) [reply R/hubData-package.R:1]: Indeed it's autogenerated by `usethis::use_lifecycle()` and allows us to display lifecycle badges on function help pages, e.g.: https://668badf304e1c106619417b2--hubdata-pr-preview.netlify.app/reference/create_model_out_submit_tmpl

**Pull request merged**